src/event.js

```diff
@@ -298,7 +298,7 @@ jQuery.event = {
 
 		var i, j, ret, matched, handleObj, handlerQueue,
 			args = new Array( arguments.length ),
-			handlers = ( dataPriv.get( this, "events" ) || {} )[ event.type ] || [],
+			handlers = dataPriv.get( this, "events" )[ event.type ] || [],
```

Comment:
I'm depending here on the fact `jQuery.event.dispatch` (where this code resides) is only called from within `eventData.handle` of the element which means the element already has private `events` data set. Let me know if you think it's still needed for some reason; it'd be good to write a test then.

Reply:
At some point `jQuery.event.dispatch` would have been called from internal functions like [jQuery.event.simulate](https://github.com/jquery/jquery/blob/17eeebb3c04c72e5d00520c191cce858844aedec/src/event.js#L605-L626) in the 1.7.x era to support oldIE. The tests for that were pulled out though, so I suspect this guard isn't needed anymore.

Reply:
@dmethvin Do you think we could remove `jQuery.event.simulate` in 4.0? The whole `jQuery.event` is not a documented API but I'm not sure which parts of it are used outside; certainly `jQuery.event.special` is. `jQuery.event.simulate` was only used internally for simulating `focusin`/`focusout` but we removed the patch in #4362.

Reply:
Note to self: check jQuery Mousewheel, in particular:
https://github.com/jquery/jquery-mousewheel/blob/aba08a57e4816c93f639ddc6c95088316b3ceb24/jquery.mousewheel.js#L218

Reply:
I'm good with keeping this change and revisiting if it causes problems.

Reply:
I restored the safeguard, just changing the fallback to `Object.create( null )`. With other changes I had to make, restoring this safeguard actually _reduced_ the gzipped size by 1 byte so why even risk. 🤷‍♂